include/InsetOrderOptimizer.h

```diff
@@ -106,77 +100,42 @@ class InsetOrderOptimizer
     const ZSeamConfig& z_seam_config;
     const std::vector<VariableWidthLines>& paths;
     const unsigned int layer_nr;
-    
-    bool added_something;
-    bool retraction_region_calculated; //Whether the retraction_region field has been calculated or not.
+
     std::vector<std::vector<ConstPolygonPointer>> inset_polys; // vector of vectors holding the inset polygons
     Polygons retraction_region; //After printing an outer wall, move into this region so that retractions do not leave visible blobs. Calculated lazily if needed (see retraction_region_calculated).
 
+    /* Determine if the paths should be reversed
+     * If there is one extruder used, and we're currently printing the inner walls then Reversing the insets now depends on the inverse of
+     * the inset direction. If we want to print the outer insets first we start with the lowest and move forward otherwise we start with the
+     * highest and iterate back.
+     * Otherwise, if the wall is partially printed with the current extruder we need to move forward for the outer wall extruder and iterate
+     * back for the inner wall extruder
+     *
+     * \param use_one_extruder boolean stating that we are using a single extruder.
+     * \param current_extruder_is_wall_x boolean stating if the current extruder is used for the inner walls.
+     * \param outer_to_inner boolean which should be set to true if we're printing from an outside to the inside
+     *
+     * \return a bool if the paths should be printed in reverse or not
+     */
```

Comment:
The comment above is good but these params and return information should be at the function declaration. With a short comment.

Reply:
https://github.com/Ultimaker/CuraEngine/pull/1761/commits/8acbcd26935d28b82aa36a8484d6a28c3a9eef53